src/react/backend/location/Locations.tsx

```diff
@@ -55,10 +44,70 @@ function Locations() {
   const endpoints = useSelector(
     (state: AppState) => state.configuration.latest.endpoints,
   );
-  const data = useMemo(() => Object.values(locations), [locations]);
-  const showDeleteLocationName = useSelector(
-    (state: AppState) => state.uiLocations.showDeleteLocation,
+  const [showModal, setShowModal] = useState(false);
+  const deleteLocationMutation = useMutation(
+    (locationName: string) => {
+      return dispatch(deleteLocation(locationName));
+    },
+    {
+      onError: () => {
+        const str = 'An error occurred during location deletion.';
+        dispatch(handleErrorMessage(str, 'byModal'));
+      }
+    },
+
   );
+  return (
+    <div>
+      <DeleteConfirmation
+        show={showModal}
+        cancel={() => setShowModal(false)}
+        approve={() => {
+          deleteLocationMutation.mutate(locationName)
+        }}
+        titleText={`Delete User Key? \n
+                    Permanently remove the following Key ${locationName} ?`}
+      />
+      <InlineButton
+        icon={<i className="far fa-edit" />}
+        variant="secondary"
+        onClick={() => history.push(`/locations/${locationName}/edit`)}
+        type="button"
+        tooltip={{
+          overlay: 'Edit Location',
+          placement: 'top',
+        }}
+        disabled={!canEditLocation(locationName, locations)}
+      />
+      <InlineButton
+        icon={<i className="fas fa-trash" />}
+        variant="danger"
+        onClick={() => setShowModal(true)}
+        type="button"
+        tooltip={{
+          overlay: 'Delete Location',
+          placement: 'top',
+        }}
+        disabled={!canDeleteLocation(locationName, locations, replicationStreams, buckets, endpoints)}
+      />
+    </div>
+  );
+};
+
+function Locations() {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const locations = useSelector(
+    (state: AppState) => state.configuration.latest.locations,
+  );
+  const replicationStreams = useSelector(
+    (state: AppState) => state.workflow.replications,
+  );
+  const buckets = useSelector((state: AppState) => state.stats.bucketList);
+  const endpoints = useSelector(
+    (state: AppState) => state.configuration.latest.endpoints,
+  );
+  const data = useMemo(() => Object.values(locations).map(location => ({...location, asyncMetadataUpdates: true, actions: true})), [locations]);
```

Comment:
Maybe we should prefix-sufix these 2 additional keys to avoid any collision with location properties
```suggestion
  const data = useMemo(() => Object.values(locations).map(location => ({...location, _asyncMetadataUpdatesColumn: true, _actionsColumn: true})), [locations]);
```